Feature/authentication user management

THERESE07 opened the pull request:
## Description
This PR implements a complete authentication system for the MediConnect API, including JWT-based authentication, Role-Based Access Control (RBAC), and comprehensive user management functionality.

### Key Features Added:
- **JWT Authentication**: Secure token-based authentication with access and refresh tokens
- **Role-Based Access Control (RBAC)**: User roles (Admin, Doctor, Patient) with proper authorization middleware
- **User Management**: Complete user registration, login, profile management, and password reset functionality
- **Security Features**: Password hashing with bcrypt, input validation, and secure token handling
- **Database Models**: Enhanced User model with proper relationships and validation
- **API Endpoints**: RESTful authentication and user management endpoints

### Changes Include:
- Authentication service with login/register/refresh token logic
- Authorization middleware for protecting routes
- User model with role-based permissions
- Password hashing and comparison utilities
- JWT token generation and validation
- Database seeders for initial user data
- Comprehensive error handling and validation

## Type of change
- [x] New feature
- [x] Bug fix
- [x] Breaking change
- [x] Documentation update

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Testing
- [x] Registration endpoint works correctly
- [x] Login endpoint generates valid JWT tokens
- [x] Password hashing and comparison functions properly
- [x] Role-based authorization middleware blocks unauthorized access
- [x] Token refresh mechanism works as expected
- [x] Database models and relationships are properly configured


## Additional Notes
- All sensitive data (passwords) are properly hashed before storage
- JWT tokens have appropriate expiration times
- Environment variables are used for sensitive configuration
- Database migrations and seeders are included for easy setup

THERESE07 committed:
feat: Complete authentication system with JWT, RBAC, and user management

THERESE07 committed:
fix: removing duplicate keys

fab-ryan commented on package.json:
```diff
@@ -34,29 +34,22 @@
     "pg": "^8.11.3",
     "pg-hstore": "^2.3.4",
     "qrcode": "^1.5.3",
-    "sequelize": "^6.35.0",
```
why did you remove sequelize here

THERESE07 replied:
My bad!
This was unintentional. It's a typo

fab-ryan commented on src/middleware/auth.ts:
```diff
@@ -0,0 +1,90 @@
+export const requireDoctor = requireRole(UserRole.DOCTOR);
+export const requirePharmacist = requireRole(UserRole.PHARMACIST);
+export const requireAdmin = requireRole(UserRole.ADMIN);
+export const requireStaff = requireRole([UserRole.DOCTOR, UserRole.PHARMACIST, UserRole.ADMIN]);
```
I need new line here

THERESE07 replied:
Sure!

fab-ryan commented on src/server.ts:
```diff
@@ -14,6 +15,9 @@ const PORT = process.env['PORT'] || 3000;
 app.use(express.json());
 
+// Mount auth routes
+app.use('/api/v1/auth', authRoutes);
```
create one file that will server as single router files

THERESE07 replied:
Suree!

fab-ryan reviewed (commented):
Please add all required environment variables in env example

THERESE07 committed:
feat: implement centralized route management system

- Add main router index to consolidate auth, patient, and swagger routes
- Refactor route files to include path prefixes (/auth, /patients)
- Update server to use centralized routing with /v1 API versioning
- Maintain existing middleware, authentication, and role-based access controls
- Enable cleaner route organization and easier maintenance

THERESE07 committed:
feat: add comprehensive unit tests for authentication system

cedrick13bienvenue reviewed (commented):
Looks good to me

THERESE07 committed:
feat: add comprehensive unit tests for authentication system

THERESE07 committed:
docs: Add comprehensive Swagger API documentation
- Add Swagger UI and JSDoc dependencies
- Create organized Swagger folder structure with paths and schemas
- Document all authentication endpoints with detailed schemas
- Include common schemas for errors and  success responses
- Add Swagger routes for API documentation access
- Update server.ts to include Swagger documentation endpoints
- Provide detailed API documentation at /api/v1/docs
